cpp/src/aztec/CMakeLists.txt

```diff
@@ -160,4 +166,13 @@ else()
         $<TARGET_OBJECTS:stdlib_merkle_tree_objects>
         $<TARGET_OBJECTS:env_objects>
     )
-endif()
\ No newline at end of file
+
+    if(INSTALL_BARRETENBERG)
+        install(
+            TARGETS barretenberg barretenberg_headers
+            EXPORT barretenbergTargets
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            FILE_SET HEADERS DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        )
```

Comment:
Could you add a comment explaining what this is doing (especially for EXPORT and each DESTINATION arg). I had to look it up and I'm sure others will too. Still not 100% sure I know what this command is doing after doing a bit of digging.

Reply:
Will do in a follow up PR.